src/shared/extensionUtilities.ts

```diff
@@ -16,6 +16,21 @@ import { getLogger } from './logger'
 
 const localize = nls.loadMessageBundle()
 
+const VSCODE_APPNAME = 'Visual Studio Code'
+
+export enum IDE {
+    vscode,
+    unknown,
+}
+
+export function getIdeType(): IDE {
+    if (vscode.env.appName.startsWith(VSCODE_APPNAME)) {
```

Comment:
Add a null check to appName to prevent unexpected error

Reply:
On one hand, this is guaranteed by the Typescript type, on the other hand, I know your firsthand experiences with Theia.